addons/pos_razorpay/models/pos_payment_method.py

```diff
@@ -1,22 +1,16 @@
 from odoo.exceptions import UserError
-from odoo import fields, models, api, _
+from odoo import api, models, _
 
 from .razorpay_pos_request import RazorpayPosRequest
 
 class PosPaymentMethod(models.Model):
     _inherit = 'pos.payment.method'
 
-    razorpay_tid = fields.Char(string='Razorpay Device Serial No', help='Device Serial No \n ex: 7000012300')
-    razorpay_allowed_payment_modes = fields.Selection(selection=[('all', 'All'), ('card', 'Card'), ('upi', 'UPI'), ('bharatqr', 'BHARATQR')], default='all', help='Choose allow payment mode: \n All/Card/UPI or QR')
-    razorpay_username = fields.Char(string='Razorpay Username', help='Username(Device Login) \n ex: 1234500121')
-    razorpay_api_key = fields.Char(string='Razorpay API Key', help='Used when connecting to Razorpay: https://razorpay.com/docs/payments/dashboard/account-settings/api-keys/')
-    razorpay_test_mode = fields.Boolean(string='Razorpay Test Mode', default=False, help='Turn it on when in Test Mode')
-
     def _get_payment_terminal_selection(self):
         return super()._get_payment_terminal_selection() + [('razorpay', 'Razorpay')]
 
     def razorpay_make_payment_request(self, data):
-        razorpay = RazorpayPosRequest(self)
+        razorpay = RazorpayPosRequest(self.pos_payment_provider_id, self.terminal_id)
```

Comment:
I would pass self and use other fields from self.
i.e. `razorpay = RazorpayPosRequest(self)`